modules/vulnerabilities/unix/web_training/security_shepherd/secgen_metadata.xml

```diff
@@ -18,13 +18,19 @@
   <platform>unix</platform>
 
   <read_fact>unix_username</read_fact>
+  <read_fact>modules</read_fact>
+  <read_fact>flag_store</read_fact>
 
   <default_input into="unix_username">
     <generator type="username_generator" />
   </default_input>
 
+  <default_input into="modules">
+    <generator type="ss_modules" />
+  </default_input>
+
   <default_input into="flag_store">
-    <generator type="flag_generator" />
+    <generator type="flag_counter" />
```

Comment:
This was (correctly) removed as a type. Use module_path=".*flag_words" if you want to specify which one. Is there a reason it needs to be that one specifically, and not any flag generator?

Reply:
Not all flag_generator types have a "number_of_words" counter which in turn is causing clashes with other flag generators (This was my original reasoning for creating a new type - although rightfully this shouldnt be needed). Thus failing the build. 

The only way around i can think around this is updating the base string generator ruby class to include the "number_of_words" argument, which in turn would allow the other flag generators access to the parameter.

Reply:
But why is that needed? The flags are just being served up by the additional SS step that TV added, so why does the format of the flag matter?

Reply:
This does not change the format of the flag. 

Just allows the generation of x amount of flags. There were two reasons for me adding this:

1. There was no way to generate a number of flags without calling the generator x times. 
2. Correlate a number of flags with the amount of modules generated by the Shepherd Module generator.

Reply:
Hmm, honestly I'm not sure the code does what you say it does. It looks like it's generating a flag string based on a specified number of characters. For example, 4 words = `flag{correcthorsebatterystaple}` 2 words= `flag{correcthorse}`

We could add that feature, but I think it's fine to just feed through as many flags as modules to enable, with multiple lines of flag generators.

Reply:
Here is the output from said changes:

![image](https://github.com/cliffe/SecGen/assets/74184849/add2178c-2c05-4a92-92bf-68b3b0c99c57)